tests/std/tests/P0433R2_deduction_guides/test.cpp

```diff
@@ -328,6 +332,82 @@ void test_function_wrapper() {
 
     static_assert(is_same_v<decltype(f9), F<double(const double&, const double&)>>);
     static_assert(is_same_v<decltype(f10), F<int(int, int)>>);
+
+#ifdef HAS_EXPLICIT_THIS_PARAMETER
+    struct ExplicitThisByVal {
+        void operator()(this ExplicitThisByVal, char) {}
+    };
+
+    ExplicitThisByVal explicit_this_by_val_functor{};
+
+    F f11(explicit_this_by_val_functor);
+    F f12(as_const(explicit_this_by_val_functor));
+    F f13(move(explicit_this_by_val_functor));
+    F f14(move(as_const(explicit_this_by_val_functor)));
+
+    static_assert(is_same_v<decltype(f11), F<void(char)>>);
+    static_assert(is_same_v<decltype(f12), F<void(char)>>);
+    static_assert(is_same_v<decltype(f13), F<void(char)>>);
+    static_assert(is_same_v<decltype(f14), F<void(char)>>);
+
+    struct ExplicitThisByRef {
+        void operator()(this ExplicitThisByRef&, short) {}
+    };
+
+    ExplicitThisByRef explicit_this_by_ref_functor{};
+
+    F f15(explicit_this_by_ref_functor);
+
+    static_assert(is_same_v<decltype(f15), F<void(short)>>);
+
+    struct ExplicitThisByCRef {
+        void operator()(this const ExplicitThisByCRef&, int) {}
+    };
+
+    ExplicitThisByCRef explicit_this_by_cref_functor{};
+
+    F f16(explicit_this_by_cref_functor);
+    F f17(as_const(explicit_this_by_cref_functor));
+    F f18(move(explicit_this_by_cref_functor));
+    F f19(move(as_const(explicit_this_by_cref_functor)));
+
+    static_assert(is_same_v<decltype(f16), F<void(int)>>);
+    static_assert(is_same_v<decltype(f17), F<void(int)>>);
+    static_assert(is_same_v<decltype(f18), F<void(int)>>);
+    static_assert(is_same_v<decltype(f19), F<void(int)>>);
+
+    struct ExplicitThisByRRef {
+        void operator()(this const ExplicitThisByRRef&, long) {}

```

Comment:
```suggestion
        void operator()(this ExplicitThisByRRef&&, long) {}
```

Reply:
Oh, I was basically wrong, because non-lvalue-callable functors are not supported ([[func.wrap.func.con]/26](https://eel.is/c++draft/func.wrap.func.con#26)).

I'll remove this case.